### PR TITLE
Added the class structure to the model file

### PR DESCRIPTION
**Pull request by akaashsidhu**

### CODE REVIEW

File: project-folder/model.py [hunk 1]
```diff
@@ -1,53 +1,86 @@
...
         self.X: np.array
         self.y: np.array
 
         self.model: Sequential
 
+    def split(self):
+        self.X_train, self.X_test, self.y_train, self.y_test =
+        train_test_split(self.X, self.y, test_size=0.20, random_state=42)
+        self.X_train, self.X_validate, self.y_train, self.y_test
+        = train_test_split(self.X_train, self.y_train, test_size=0.20,
```
Comment (happilyeverafter95) [inline project-folder/model.py]: I think that's too much data given to test/validation, esp since validation is only used to tune the learning rate. Training data is only 64% here. I would suggest no more than 10% for validation.

File: project-folder/model.py [hunk 2]
```diff
@@ -1,53 +1,86 @@
...
-        model.add(Dense(128, activation='relu'))
-        model.add(Dropout(0.3))
-        model.add(Dense(self.num_classes, activation='softmax'))
+        model.add(Dense(512, activation='relu'))
+        model.add(Dropout(0.55))
+        model.add(Dense(7, activation='softmax'))
 
-        model.compile(loss='categorical_crossentropy',
-                      optimizer='adam',
+        model.compile(optimizer='adam', loss='binary_crossentropy',
```
Comment (happilyeverafter95) [inline project-folder/model.py]: categorical_crossentropy

File: project-folder/model.py [hunk 3]
```diff
@@ -1,53 +1,86 @@
...
-        self.y = np.eye(self.num_classes)[np.array(y.astype(int)).reshape(-1)]
-
     # Entry point to training the model
     # Consumes data from DataPipeline
     def model_runner(self, data):
         self.reshape_data(data)
+        self.datagen.fit(self.X_train)
         self.model = self.define_model()
-        self.model.fit(self.X, self.y,
+        self.model.fit(self.X_train, self.y_train,
```
Comment (happilyeverafter95) [inline project-folder/model.py]: you should be using your datagen here

File: project-folder/model.py [hunk 4]
```diff
@@ -1,53 +1,86 @@
...
                        batch_size=self.batch_size,
-                       epochs=self.epochs)
+                       steps_per_epoch=self.X_train.shape[0] //
```
Comment (happilyeverafter95) [inline project-folder/model.py]: you don't need to specify this. The default is number of training data / batch size

File: project-folder/CNN.ipynb [hunk 5]
```diff
@@ -2,23 +2,28 @@
  "cells": [
   {
    "cell_type": "code",
-   "execution_count": 121,
+   "execution_count": 1,
```
Comment (happilyeverafter95) [inline project-folder/CNN.ipynb]: Let's find out next time why this file keeps getting added :P

File: project-folder/model.py (hunk 3 above)
Comment (akaashsidhu) [reply project-folder/model.py]: Should it be: self.model.fit(self.datagen.fit(self.X_train), self.y_train, then?
